File: inspect-excel.mjs
#!/usr/bin/env node

import ExcelJS from 'exceljs';

const filePath = process.argv[2];

if (!filePath) {
  console.error('Uso: node inspect-excel.mjs <ruta-archivo>');
  process.exit(1);
}

async function inspectExcel() {
  try {
    const workbook = new ExcelJS.Workbook();
    await workbook.xlsx.readFile(filePath);

    console.log('\n' + '='.repeat(80));
    console.log('📊 ESTRUCTURA DEL EXCEL DE PEDRO');
    console.log('='.repeat(80));

    for (const worksheet of workbook.worksheets) {
      console.log(`\n📄 HOJA: ${worksheet.name}`);
      console.log('-'.repeat(80));

      worksheet.eachRow({ includeEmpty: false }, (row, rowNum) => {
        console.log(`\nFila ${rowNum}:`);
        row.eachCell({ includeEmpty: false }, (cell, colNum) => {
          const colLetter = String.fromCharCode(64 + colNum);
          console.log(
            `  ${colLetter}${rowNum}: "${cell.value}" ` +
              `| Font: ${cell.font?.bold ? 'BOLD' : 'normal'}, ` +
              `Size: ${cell.font?.size || 11}, ` +
              `Fill: ${cell.fill?.fgColor?.rgb || 'none'}, ` +
              `Align: ${cell.alignment?.horizontal || 'default'}, ` +
              `Merged: ${cell.isMerged ? 'YES' : 'NO'}`
          );
        });
      });
    }

    console.log('\n' + '='.repeat(80));
    console.log('✅ Análisis completado');
    console.log('='.repeat(80) + '\n');
  } catch (error) {
    console.error('❌ Error:', error.message);
    process.exit(1);
  }
}

inspectExcel();

File: inspect-excel.py
#!/usr/bin/env python3
import openpyxl
import sys

try:
    wb = openpyxl.load_workbook(sys.argv[1])

    print("\n" + "="*70)
    print("ESTRUCTURA DEL EXCEL")
    print("="*70)

    for sheet_name in wb.sheetnames:
        ws = wb[sheet_name]
        print(f"\n📄 HOJA: {sheet_name}")
        print("-"*70)

        max_row = 0
        for row in ws.iter_rows():
            if any(cell.value for cell in row):
                max_row = row[0].row

        print(f"Total filas con datos: {max_row}")
        print("\nContenido:")

        for row_num in range(1, min(max_row + 1, 100)):
            row_data = []
            for col_num in range(1, 10):
                cell = ws.cell(row_num, col_num)
                if cell.value:
                    row_data.append(f"Col{col_num}:'{cell.value}'")

            if row_data:
                print(f"Fila {row_num:2d}: {' | '.join(row_data)}")

except Exception as e:
    print(f"Error: {e}")
    import traceback
    traceback.print_exc()
